src/project_workspace.ts

```diff
@@ -69,15 +76,17 @@ export default class ProjectWorkspace {
 
   constructor(
     rootPath: string,
-    pathToJest: string,
-    pathToConfig: string,
+    jestCommandLine: string,
     localJestMajorVersion: number,
+    pathToConfig?: string,
```

Comment:
I remember why I wanted to change the order of these arguments.  Its because the `pathToConfig` property is genuinely optional.  It is checked before it is used and it makes sense that there are occasions when there is no config file - just use the Jest defaults.  For the purposes of this PR, I have reverted the constructor to maintain backwards compatibility.  I will probably aim to tackle this scenario in the TS conversion PR.  I personally think that we should use an interface for this - which is easier to do in TS.

To work around this issue in my code I do this: `pathToConfig: undefined as unknown as string`.

Reply:
Just a thought -- If we're having to worry about argument order, that signals to me we should consider using an Object.

Reply:
Yes, I agree that an object would be the best option here.  But my opinion is that it will be easier to make that transition once more of the code is converted to Typescript.  I just played with defining an interface for the ProjectWorkspace class, and then realized that I would need to update some existing code in JS with Flow typings...

Reply:
actually come to think about this, why is this a class? what does it do other than holding some config data? maybe in the next major version, possible typescript conversion (?), we can rewrite this...

Reply:
Sorry to derail that thread. I've created #46 to capture that idea. We can defer that for now.

To get back to the original point, the API is messy. I'd say leave the argument order for now and defer changing the API signature. In the interim Ross, you could build a factory function that only has to add that `pathToConfig: undefined as unknown as string` mess once.

```js
const workspace = ({ ... }) => new ProjectWorkspace( ... );
```